ddtrace/appsec/processor.py

```diff
@@ -293,15 +361,15 @@ def on_span_finish(self, span):
 
             span.set_metric(APPSEC_EVENT_RULE_LOADED, info.loaded)
             span.set_metric(APPSEC_EVENT_RULE_ERROR_COUNT, info.failed)
-            if ddwaf_result:
-                span.set_metric(APPSEC_WAF_DURATION, ddwaf_result.runtime)
-                span.set_metric(APPSEC_WAF_DURATION_EXT, ddwaf_result.total_runtime)
+            if not blocked_request:
+                span.set_metric(APPSEC_WAF_DURATION, total_runtime)
+                span.set_metric(APPSEC_WAF_DURATION_EXT, total_overall_runtime)
         except JSONDecodeError:
             log.warning("Error parsing data AppSec In-App WAF metrics report")
         except Exception:
             log.warning("Error executing AppSec In-App WAF metrics report: %s", exc_info=True)
 
-        if ddwaf_result and ddwaf_result.data is not None:
+        if blocked_request or (ddwaf_result and ddwaf_result.data is not None):
```

Comment:
is there a way we can move this block of code earlier?

I think it is a rather jarring block to parse.

e.g.

```python
blocked_request = _context.get_item("http.request.blocked", span=span)
if not blocked_request:
    # run the waf again
    set_metric(APPSEC_WAF_DURATION, ...)
    log.debug(DDAS-011-00])
    set_tag_str(APPSEC_JSON, ddwaf_result.data)
else:
    set_tag(APPSEC_JSON, _context.get_item())
    set_tag("appsec.blocked", True)


try:
    info = self._ddwaf_info
except:
    pass
```

I am not convinced this is the best suggestion to structure it, but I definitely find myself scrolling back and forth a bunch to try and follow the paths since there are a few places where we check both blocked_request and ddwaf_result

Reply:
Simplified the logic and added a boolean for either first or second ddwaf runs returning true which should make this more readable.